hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/IncrementalTableBackupClient.java

```
@@ -249,9 +249,15 @@ private void updateFileLists(List<String> activeFiles, List<String> archiveFiles
       }
     }
 
-    if (newlyArchived.size() > 0) {
+    if (!newlyArchived.isEmpty()) {
+      String rootDir = CommonFSUtils.getRootDir(conf).toString();
+
       activeFiles.removeAll(newlyArchived);
-      archiveFiles.addAll(newlyArchived);
+      for (String file : newlyArchived) {
+        String archivedFile = file.substring(rootDir.length() + 1);
+        archivedFile = new Path(HFileArchiveUtil.getArchivePath(conf), archivedFile).toString();
```

Should you confirm that this new file path exists before handing it off to the next step? Maybe complain loudly if the file doesn't exist in the new or old location?

Just thinking about how this might silently drop data due to a path manipulation bug.

---

We won't drop data b/c the call to backup a file that doesn't exist will fail, but I can probably just add a check here to fast fail. Thanks